### DISCUSSION TITLE
macOS support

### DISCUSSION THREAD

**Discussion by ben-cl** [bloc97/Anime4K#179] [Q&A]
:)Hi! 
I love the project. Works well on my PC. Content de voir que L'UdeM à contribué (Je suis de Montréal aussi hehe). If I follow the Linux installation instruction, will it work on macOS? 
I have the m1 chips if that changes anything. 

Thanks!

**Comment by bloc97** [comment]
The linux installation instruction should work, but I vaguely remember something about Apple deprecating OpenGL on the M1 chips? If there are problems you could try using Vulkan instead of OpenGL. https://mpv.io/manual/stable/#options-gpu-api

**Comment by ben-cl** [comment]
:)Thanks for your feedback, it is working! I did spend some hours trying to figure out the best way. 

First I did install mpv with the help of this reddit post. It works but it is a challenge to get native build for ARM: https://www.reddit.com/r/mpv/comments/qt41pl/how_to_build_mpv_for_m1_macs_arm64/

After more research, the best way I found is to use iina : https://github.com/iina/iina
The player is base on mpv, but much easier to install and have a modern look. 
First you need in preferences - Advanced, to Enable advanced settings with the Use config directory. 
<img width="1106" alt="image" src="https://user-images.githubusercontent.com/32286421/173250447-dcbfd905-6ae2-4b75-b1f6-af02add9d188.png">
We add the shaders folder as usual. 
After in preferences again we can import the input.conf file inside Key Bindings, with Import an existing config file. 
<img width="1106" alt="image" src="https://user-images.githubusercontent.com/32286421/173250586-7e21e20d-4e25-4a6b-97ac-ecd557566bb3.png">

Strange to say but I haven't yet contributed to an open source project. If I could create a pull request with more detail instruction for Mac, I would love too!

Thanks again

**Comment by yooooooLeon** [comment]
您的邮件已收到！谢谢！
